packages/api/src/beacon/routes/node.ts

```diff
@@ -77,148 +109,198 @@ export type NodeHealthOptions = {
+      },
+      meta: EmptyMetaCodec,
+    },
+  },
+  getPeers: {
+    url: "/eth/v1/node/peers",
+    method: "GET",
+    req: {
+      writeReq: ({state, direction}) => ({query: {state, direction}}),
+      parseReq: ({query}) => ({state: query.state, direction: query.direction}),
       schema: {query: {state: Schema.StringArray, direction: Schema.StringArray}},
     },
-    getPeer: {
-      writeReq: (peer_id) => ({params: {peer_id}}),
-      parseReq: ({params}) => [params.peer_id],
-      schema: {params: {peer_id: Schema.StringRequired}},
+    resp: {
+      ...JsonOnlyResponseCodec,
+      meta: {
+        toJson: (d) => d,
+        fromJson: (d) => ({count: (d as PeersMeta).count}),
+        toHeadersObject: () => ({}),
+        fromHeaders: () => ({}) as PeersMeta,
+      },
+      transform: {
+        toResponse: (data, meta) => ({data, meta}),
+        fromResponse: (resp) => resp as {data: NodePeer[]; meta: PeersMeta},
+      },
     },
-
-    getPeerCount: reqEmpty,
-    getNodeVersion: reqEmpty,
-    getSyncingStatus: reqEmpty,
-    getHealth: {
-      writeReq: (options) => ({
-        query: options?.syncingStatus !== undefined ? {syncing_status: options.syncingStatus} : {},
-      }),
-      parseReq: ({query}) => [{syncingStatus: query.syncing_status}],
-      schema: {query: {syncing_status: Schema.Uint}},
+  },
+  getPeer: {
+    url: "/eth/v1/node/peers/{peer_id}",
+    method: "GET",
+    req: {
+      writeReq: ({peerId}) => ({params: {peer_id: peerId}}),
+      parseReq: ({params}) => ({peerId: params.peer_id}),
+      schema: {params: {peer_id: Schema.StringRequired}},
     },
-  };
-}
-
-export function getReturnTypes(): ReturnTypes<Api> {
-  const stringType = new StringType();
-  const NetworkIdentity = new ContainerType(
-    {
-      peerId: stringType,
-      enr: stringType,
-      p2pAddresses: ArrayOf(stringType),
-      discoveryAddresses: ArrayOf(stringType),
-      metadata: ssz.altair.Metadata,
+    resp: JsonOnlyResponseCodec,
+  },
+  getPeerCount: {
+    url: "/eth/v1/node/peer_count",
+    method: "GET",
+    req: EmptyRequestCodec,
+    resp: {
+      data: PeerCountType,
+      meta: EmptyMetaCodec,
     },
-    {jsonCase: "eth2"}
-  );
-
-  const PeerCount = new ContainerType(
-    {
-      disconnected: ssz.UintNum64,
-      connecting: ssz.UintNum64,
-      connected: ssz.UintNum64,
-      disconnecting: ssz.UintNum64,
+  },
+  getNodeVersion: {
+    url: "/eth/v1/node/version",
+    method: "GET",
+    req: EmptyRequestCodec,
+    resp: JsonOnlyResponseCodec,
+  },
+  getSyncingStatus: {
+    url: "/eth/v1/node/syncing",
+    method: "GET",
+    req: EmptyRequestCodec,
+    resp: JsonOnlyResponseCodec,
+  },
+  getHealth: {
+    url: "/eth/v1/node/health",
+    method: "GET",
+    req: {
+      writeReq: ({syncingStatus}) => ({query: {syncing_status: syncingStatus}}),
+      parseReq: ({query}) => ({syncingStatus: query.syncing_status}),
+      schema: {query: {syncing_status: Schema.Uint}},
```

Comment:
`Uint` have different range but this parameter have different integer range `100-599`.

Reply:
This is the same as on unstable, `Schema.Unit` will currently be translated to `{type: "integer", minimum: 0};`, we can add more accurate / ranged-based schema validation for integer values but this is the only API as far as I know that requires it.

We currently already enforce that the range is valid in server implementation
https://github.com/ChainSafe/lodestar/blob/312136374e254a58e7b16e6c639cd0a579231add/packages/beacon-node/src/api/impl/node/index.ts#L72-L73

Reply:
Keeping as is, can be addressed separately from this PR. @nazarhussain if you think it's better to validate this as part of the schema instead of server implementation, we can open a separate PR (or just an issue) for this